multiqc/modules/ngsderive/ngsderive.py

```diff
@@ -124,37 +112,61 @@ def probe_file_for_dictreader_kwargs(self, f, expected_header_count):
         f.seek(0)
 
         for delim in ["\t", ",", " ", "|"]:
-            if delim in header and len(header.split(delim)) == expected_header_count:
+            if delim in header and len(header.split(delim)) in expected_header_counts:
                 return {"delimiter": delim}
 
         log.warn("Could not easily detect delimiter for file. Trying csv.Sniffer()")
         dialect = csv.Sniffer().sniff(f.read(4096))  # 4096 is the buffer size
         f.seek(0)
         return {"dialect": dialect}
 
-    def parse(self, sample_dict, found_file, subcommand, expected_header_count):
-        kwargs = self.probe_file_for_dictreader_kwargs(io.StringIO(found_file["f"]), expected_header_count)
+    def parse(self, sample_dict, found_file, subcommand, expected_header_counts):
+        kwargs = self.probe_file_for_dictreader_kwargs(io.StringIO(found_file["f"]), expected_header_counts)
         for row in csv.DictReader(io.StringIO(found_file["f"]), **kwargs):
             if not row.get("File"):
                 continue
             sample_name = self.clean_s_name(row.get("File"), found_file)
+            read_group = row.get("ReadGroup")
+            if read_group and read_group != "overall":
+                sample_name += "-" + read_group
+                row["is_read_group"] = True
+            else:
+                row["is_read_group"] = False
             if sample_name in sample_dict:
-                log.debug("Duplicate sample name found! Overwriting: {}".format(sample_name))
+                log.debug(f"Duplicate sample name found for {subcommand}! Overwriting: {sample_name}")
 
             sample_dict[sample_name] = row
-            self.add_data_source(f=found_file, s_name=sample_name)
+            if not row["is_read_group"]:
+                self.add_data_source(f=found_file, s_name=sample_name)
 
     def add_strandedness_data(self):
         # Write data to file
         self.write_data_file(self.strandedness, "ngsderive_strandedness")
 
         data = {}
         for sample, strandedness in self.strandedness.items():
-            data[sample] = {
-                "predicted": strandedness.get("Predicted"),
-                "forward": round(float(strandedness.get("ForwardPct")) * 100.0, 2),
-                "reverse": round(float(strandedness.get("ReversePct")) * 100.0, 2),
-            }
+            if float(strandedness.get("ForwardPct")) == 0 and float(strandedness.get("ReversePct")) == 0:
+                # TODO how to present this case? No reads found for this Read Group.
```

Comment:
@vladsavelyev This is a case I'm stumped over. In v4 of ngsderive, I added a feature where the read group definitions in the BAM header are validated against what appears in the sequences. This now means it's possible for an entry to be NULL, if the read group is defined in the header but lacks any sequences associated with it. There's no data to present here, so it can't go into the barplot. This is most likely an error in the BAM that should be communicated to the user. However I'm not sure how to go about that. I suppose we could `log.warning("Read Group info messed up! Yada yada")` and just ignore it for the output report... I would like to see it somewhere in the report, but I have no idea about an appropriate way to achieve that.